python_modules/libraries/dagster-airbyte/dagster_airbyte/asset_defs.py

```diff
@@ -88,3 +93,209 @@ def _assets(context):
+        ),
+        stream_data=config_contents.get("sync_catalog", {}).get("streams", []),
+    )
+
+
+def _airbyte_source_from_config(contents: Mapping[str, Any]) -> AirbyteSource:
+    return AirbyteSource(name=contents["resource_name"])
+
+
+@experimental
+def load_assets_from_airbyte_project(
+    project_dir: str,
+    key_prefix: Optional[CoercibleToAssetKeyPrefix] = None,
+    source_key_prefix: Optional[CoercibleToAssetKeyPrefix] = None,
+    create_assets_for_normalization_tables: bool = True,
+    connection_to_group_fn: Optional[Callable[[str], Optional[str]]] = _clean_name,
+) -> Tuple[List[AssetsDefinition], List[SourceAsset]]:
+    """
+    Loads an Airbyte project into a set of Dagster assets.
+
+    Point to the root folder of an Airbye project synced using the Octavia CLI. For
+    more information, see https://github.com/airbytehq/airbyte/tree/master/octavia-cli#octavia-import-all.
+
+    Args:
+        project_dir (str): The path to the root of your Airbyte project, containing sources, destinations,
+            and connections folders.
+        key_prefix (Optional[CoercibleToAssetKeyPrefix]): A prefix for the asset keys created.
+        source_key_prefix (Optional[CoercibleToAssetKeyPrefix]): A prefix for the source asset keys produced.
+        create_assets_for_normalization_tables (bool): If True, assets will be created for tables
+            created by Airbyte's normalization feature. If False, only the destination tables
+            will be created.
+        connection_to_group_fn (Optional[Callable[[str], Optional[str]]]): Function which returns an asset
+            group name for a given Airbyte connection name. If None, no groups will be created. Defaults
+            to a basic sanitization function.
+    """
+
+    if isinstance(source_key_prefix, str):
+        source_key_prefix = [source_key_prefix]
+    source_key_prefix = check.list_param(source_key_prefix or [], "source_key_prefix", of_type=str)
+
+    if isinstance(key_prefix, str):
+        key_prefix = [key_prefix]
+    key_prefix = check.list_param(key_prefix or [], "key_prefix", of_type=str)
+
+    assets: List[AssetsDefinition] = []
+    source_assets: Dict[str, SourceAsset] = {}
+
+    connections_dir = os.path.join(project_dir, "connections")
+    for connection_name in os.listdir(connections_dir):
+        connection_dir = os.path.join(connections_dir, connection_name)
+        with open(os.path.join(connection_dir, "configuration.yaml"), encoding="utf-8") as f:
+            connection = _airbyte_connection_from_config(yaml.safe_load(f.read()))
+
+        with open(os.path.join(project_dir, connection.source_config_path), encoding="utf-8") as f:
+            source = _airbyte_source_from_config(yaml.safe_load(f.read()))
+            if source.name not in source_assets:
+                source_asset_key = AssetKey(source_key_prefix + [_clean_name(source.name)])
+                source_assets[source.name] = SourceAsset(key=source_asset_key)
```

Comment:
I thought about this a bit more, and I'm not sure that we should create actual SourceAsset objects on behalf of the user. For the dbt case, the reason we don't create SourceAssets is that it's common for users to want to supply their own asset definitions for the data that dbt is consuming, so if we created SourceAssets on their behalf, then these definitions would conflict with the definitions that they wanted to define.

It's definitely less likely for Airbyte users to want to supply these definitions (airbyte is often the first step of a data pipeline), but it's unclear if it adds much value to provide them to the user, compared to the cost of making users understand that this function returns two separate lists.

Reply:
I do think there's some visual benefit to have a source asset present, in my opinion it makes it much more clear that the resulting assets are tied together. Without it the user is just presented with a bundle of free-floating assets that don't have as clear an indication that they come from the same origin.

<img width="927" alt="Screen Shot 2022-09-12 at 3 34 04 PM" src="https://user-images.githubusercontent.com/10215173/189770118-c674ea5d-22b6-41cc-ba9b-93cc416be41c.png">

Maybe we could output a single, mixed list as Sandy suggested above?

Reply:
In the case where each table in Snowflake is pulled from a table in the user's app DB, would we have a source asset per table in the user's app DB or a source asset for "app DB"?

Reply:
Updated to output a single list. Let me know if the new API feels ok.

It might make sense to either allow a user to suppress creating source assets or substitute fully-defined assets. I do think it is a nice default to have the source assets created, though.

Reply:
hm the presence / absence of the source asset should not change the visual representation here, to the best of my knowledge. I think the only change would be that if you clicked on the link w/o the source asset there'd be no page for it (not even confident that's true)

Reply:
> In the case where each table in Snowflake is pulled from a table in the user's app DB, would we have a source asset per table in the user's app DB or a source asset for "app DB"?

At present, the latter. It's one-source-asset-per-data-source (GitHub, a Postgres DB etc). We could instead have one-source-asset-per-"stream". That makes some more sense in the DB-to-DB sync case since the source tables would be individually identified. In the above case where we're syncing a bunch of data from GitHub, it might be a bit messier. What do you think?

Reply:
> hm the presence / absence of the source asset should not change the visual representation here

Sorry, I think i misunderstood due to a lack of understanding of the SourceAsset system. It sounds like we can have an upstream `AssetKey` specified without actually creating / returning the `SourceAsset` object? Things should still work OK then?

Reply:
> At present, the latter. It's one-source-asset-per-data-source (GitHub, a Postgres DB etc). We could instead have one-source-asset-per-"stream". That makes some more sense in the DB-to-DB sync case since the source tables would be individually identified. In the above case where we're syncing a bunch of data from GitHub, it might be a bit messier. What do you think?

I think I'd expect a source asset per stream.  If the main reason for the source assets is to group things visually, I think we should find a different way to do that.  My lean would be to punt on the source assets entirely.  Might be easiest to chat about this on zoom?

Reply:
Following offline discussion, removed upstream assets for now, with the option to add them down the line.